### ISSUE TITLE
JSDoc @type annotation ignored in chained variable declaration

### ISSUE THREAD

**Issue by chovanecm** [microsoft/TypeScript#35544]
I'm not sure whether this is a bug or a feature request:

**TypeScript Version:**  Version 3.8.0-dev.20191206
also confirmed in 3.7.2

**Search Terms:** type ignored

**Code**

```javascript
var 
/** @type {string} */
str1,
/** @type {number} */
num1;

/** @type {string} */
var str2;
/** @type {number} */
var num2;



str1 = "Hello";
num1 = "World"; // TSC does not complain => not OK

str2 = "Hello";
num2 = "World"; // TSC complains => OK
```

**Expected behavior:**
```
test.js:15:1 - error TS2322: Type '"World"' is not assignable to type 'number'.

15 num1 = "World";

test.js:18:1 - error TS2322: Type '"World"' is not assignable to type 'number'.

18 num2 = "World";
```

**Actual behavior:**
```
test.js:18:1 - error TS2322: Type '"World"' is not assignable to type 'number'.

18 num2 = "World";
```
**Playground Link:** Unable to reproduce any type related warning in JS PlayGround, please test locally

**Related Issues:** #22434, #13371 (somewhat related)

The official JSDoc documentation says nothing about mixing types within one `var` statement, but I guess that if the TS notation works, the behavior for JS might be the same.
```ts
var str1: string,
num1: number;
```

**Comment by RyanCavanaugh** [comment]
This isn't a place we look for comments, but we could add it. Honestly if there's going to be a JS Doc type annotation on each variable, I don't see why someone would write it that way -- you "waste" a vertical line on the `var` part and it looks more confusing 🤷‍♂️

**Comment by chovanecm** [comment]
@RyanCavanaugh I must admit it's a selfish reason. The client's eslint settings require chaining it into one `var` statement :-(
